src/Glpi/Controller/LegacyFileLoadController.php

```diff
@@ -55,11 +55,58 @@ public function __invoke(Request $request): Response
             throw new \RuntimeException('Cannot load legacy controller without specifying a file to load.');
         }
 
-        $callback = function () use ($target_file) {
-            require $target_file;
-        };
+        \ob_start();
+        $response = require($target_file);
+        $content = \ob_get_clean();
 
-        return new HeaderlessStreamedResponse($callback->bindTo($this, self::class));
+        if ($response instanceof Response) {
+            // The legacy file contains a return value that corresponds to a valid Symfony response.
+            // This response is returned and any output is discarded.
+
+            if ($content !== '') {
+                \trigger_error(
+                    sprintf('Unexpected output detected in `%s`.', $target_file),
+                    E_USER_WARNING
+                );
+            }
+
+            return $response;
+        }
+
+        if (\headers_sent()) {
+            // Headers are already sent, so we have to use a streamed response without headers.
+            // This may happen when `flush()`/`ob_flush()`/`ob_clean()`/`ob_end_clean()` functions are used
+            // in the legacy script. The script should be fixed to prevent this.
+
+            \trigger_error(
+                sprintf('Unexpected output detected in `%s`.', $target_file),
```

Comment:
```suggestion
                sprintf('Sending headers in scripts is deprecated. Occurred in `%s`.', $target_file),
```

Makes it clearer that it's about headers

Reply:
Headers will be sent only if something forces the output to be flushed. Otherwise, it will be buffered.